src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -0,0 +1,301 @@
+        """
+        try:
+            props = self.profiles[profile_name]["properties"]
+        except Exception as exc:
+            raise ProfileNotFound("Profile {profile_name} not found", error_msg=exc)
+
+        secure_fields: list = self.profiles[profile_name].get("secure", [])
+
+        # load secure props only if there are secure fields
+        if secure_fields:
+            self.load_secure_props()
+
+            # load properties with key as profile.{profile_name}.properties.{*}
+            for (key, value) in self.secure_props.items():
+                if re.match(
+                    "profiles\\." + profile_name + "\\.properties\\.[a-z]+", key
+                ):
+                    property_name = key.split(".")[3]
+                    if property_name in secure_fields:
+                        props[property_name] = value
+                        secure_fields.remove(property_name)
+
+            if len(secure_fields) > 0:
+                raise SecureValuesNotFound(secure_fields)
+
+        return props
+
+    def load_secure_props(self) -> None:
+        """
+        load secure_props stored for the given config
+
+        if keyring is not initialized, set empty value
+        """
+        if not HAS_KEYRING:
+            self.secure_props = {}
+            return
+
+        try:
+            service_name = constants["ZoweServiceName"]
+
+            if sys.platform == "win32":
+                service_name += "/" + constants["ZoweAccountName"]
+
+            secret_value = keyring.get_password(
+                service_name, constants["ZoweAccountName"]
+            )
+
+        except Exception as exc:
+            raise SecureProfileLoadFailed(
+                constants["ZoweServiceName"], error_msg=str(exc)
+            ) from exc
+
+        secure_config: str
+        if sys.platform == "win32":
+            secure_config = secret_value.encode("utf-16")
+        else:
+            secure_config = secret_value
+
+        secure_config_json = commentjson.loads(base64.b64decode(secure_config).decode())
+
+        # first look for credentials stored for currently loaded config
+        # then look for default credential stored for user_directory/.zowe/zowe.config.json
+        try:
+            self.secure_props = secure_config_json[self.location]
+        except KeyError:
+            try:
+                self.secure_props = secure_config_json[GLOBAL_CONFIG_PATH]
+            except KeyError as exc:
+                raise SecureProfileLoadFailed(
+                    constants["ZoweServiceName"],
+                    error_msg="No credentials found for loaded config file as well as for global config",
+                )
+            else:
+                warnings.warn(
+                    f"Credentials not found for given config, using global credentials {GLOBAL_CONFIG_PATH}",
+                    SecurePropsNotFoundWarning,
+                )
```

Comment:
I'm probably missing something, but why do we throw an error when credentials are found in the `global_config_path` ?

Reply:
We are just warning that credentials specific to that config path is not found ... because the credentials stored under global_config_path may not be appropriate to load.